edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Mqtt/MessageAddressConverter.cs

```diff
@@ -86,32 +86,38 @@ public bool TryBuildProtocolAddressFromEdgeHubMessage(string endPointUri, IMessa
         /// of the <paramref name="message"/> argument - which here is an MQTT topic name - and
         /// extracts features from it that it then uses to populate the <see cref="ProtocolGateway.Messaging.IMessage.Properties"/>
         /// property.
+        /// Note: In case of multiple matches, this method uses the first topic. So the order in which the topics are defined matters.
         /// </summary>
         public bool TryParseProtocolMessagePropsFromAddress(IProtocolGatewayMessage message)
         {
-            var uri = new Uri(message.Address, UriKind.Relative);
+            string address = message.Address;
+            var uri = new Uri(address, UriKind.Relative);
             IList<IList<KeyValuePair<string, string>>> matches = this.inboundTable
                 .Select(template => template.Match(uri))
                 .Where(match => match.Count > 0)
                 .ToList();
 
-            if (matches.Count > 1)
-            {
-                this.logger.LogWarning($"Message address {message.Address} matches more than one topic. Picking first matching topic.");
```

Comment:
Was this not actually being hit before then?

Reply:
It was, it was just not being handled properly (was being treated as an error case)